File: src/app/models/credit-card.ts
export class CreditCard {
    public get userId(): number {
        return this._userId;
    }
    public set userId(value: number) {
        this._userId = value;
    }
    public get interestRate(): number {
        return this._interestRate;
    }
    public set interestRate(value: number) {
        this._interestRate = value;
    }
    public get creditLimit(): number {
        return this._creditLimit;
    }
    public set creditLimit(value: number) {
        this._creditLimit = value;
    }

    public get currency(): string {
        return this._currency;
    }
    public set currency(value: string) {
        this._currency = value;
    }
    public get balance(): number {
        return this._balance;
    }
    public set balance(value: number) {
        this._balance = value;
    }
    public get id(): number {
        return !this._id? -1 : this._id;
    }
    public set id(value: number) {
        this._id = value;
    }

    constructor(
        private _balance: number,
        private _currency: string,
        private _userId: number,  
        private _creditLimit: number,
        private _interestRate: number,
        private _id?: number, 
        ) {}
    
}

File: Micro/src/main/java/com/example/model/Account/CreditCard.java
package com.example.model.Account;


import com.example.model.others.Money;
import com.example.model.others.User;
import org.hibernate.annotations.DynamicUpdate;

import javax.persistence.Entity;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;
import javax.validation.constraints.DecimalMax;
import javax.validation.constraints.DecimalMin;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;
import java.math.RoundingMode;
import java.util.Date;

@Entity
@Inheritance(strategy = InheritanceType.JOINED)
@DynamicUpdate
public class CreditCard extends Account {

    @DecimalMax(value = "100000")
    @DecimalMin(value = "100")
    private BigDecimal creditLimit = new BigDecimal((100));
    @DecimalMin(value = "0.1")
    @DecimalMax(value = "0.2")
    private BigDecimal interestRate = new BigDecimal((0.2)).setScale(2,RoundingMode.HALF_UP);
    private Date lastInterestDate;

    public CreditCard(){
        this.lastInterestDate = new Date();

    }

    public CreditCard(@NotNull(message = "A balance is required") Money balance,
                      @NotNull(message = "A primaryOwner is required") User user) {
        super(balance, user);
        this.lastInterestDate = new Date();
    }

    public CreditCard(@NotNull(message = "A balance is required") Money balance,
                      @NotNull(message = "A primaryOwner is required") User user,
                      Date dateOfCreation,
                      @DecimalMax(value = "100000") @DecimalMin(value = "100") BigDecimal creditLimit) {
        super(balance, user);
        this.creditLimit = creditLimit;
        this.lastInterestDate = new Date();
    }


    public CreditCard(@NotNull(message = "A balance is required")Money balance,
                      @NotNull(message = "A primaryOwner is required")User user,
                      @DecimalMin(value = "0.1") @DecimalMax(value = "0.2") BigDecimal interestRate) {
        super(balance, user);
        this.interestRate = interestRate;
        this.lastInterestDate = new Date();
    }


    public CreditCard(@NotNull(message = "A balance is required")Money balance,
                      @NotNull(message = "A primaryOwner is required")User user,
                      @DecimalMax(value = "100000") @DecimalMin(value = "100") BigDecimal creditLimit,
                      @DecimalMin(value = "0.1") @DecimalMax(value = "0.2", inclusive = false) BigDecimal interestRate) {
        super(balance, user);
        this.creditLimit = creditLimit;
        this.interestRate = interestRate;
        this.lastInterestDate = new Date();
    }

    public BigDecimal getCreditLimit() {
        return creditLimit;
    }

    public void setCreditLimit(BigDecimal creditLimit) {
        this.creditLimit = creditLimit;
    }

    public BigDecimal getInterestRate() {
        return interestRate;
    }

    public void setInterestRate(BigDecimal interestRate) {
        this.interestRate = interestRate;
    }

    public Date getLastInterestDate() {
        return lastInterestDate;
    }

    public void setLastInterestDate(Date lastInterestDate) {
        this.lastInterestDate = lastInterestDate;
    }
}
